ResearchKit/Common/ORKReviewStep.m

```diff
@@ -0,0 +1,98 @@
+ list of conditions and the following disclaimer.
+ 
+ 2.  Redistributions in binary form must reproduce the above copyright notice,
+ this list of conditions and the following disclaimer in the documentation and/or
+ other materials provided with the distribution.
+ 
+ 3.  Neither the name of the copyright holder(s) nor the names of any contributors
+ may be used to endorse or promote products derived from this software without
+ specific prior written permission. No license is granted to the trademarks of
+ the copyright holders even if such marks are included in this software.
+ 
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+ FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+
+#import "ORKReviewStep.h"
+#import "ORKReviewStep_Internal.h"
+#import "ORKHelpers.h"
+#import "ORKReviewStepViewController.h"
+
+
+@implementation ORKReviewStep
+
+- (instancetype)initWithIdentifier:(NSString *)identifier
+                             steps:(nullable NSArray *)steps
+                      resultSource:(nullable id<ORKTaskResultSource, NSSecureCoding>)resultSource {
+    self = [super initWithIdentifier:identifier];
+    if (self) {
+        _steps = [steps copy];
+        _resultSource = resultSource;
+    }
+    return self;
+}
+
++ (instancetype)standaloneReviewStepWithIdentifier:(NSString *)identifier
+                                             steps:(nonnull NSArray *)steps
+                                      resultSource:(nullable id<ORKTaskResultSource, NSSecureCoding>)resultSource {
+    return [[ORKReviewStep alloc] initWithIdentifier:identifier steps:steps resultSource:resultSource];
+}
+
++ (instancetype)embeddedReviewStepWithIdentifier:(nonnull NSString *)identifier {
+    return [[ORKReviewStep alloc] initWithIdentifier:identifier steps:nil resultSource:nil];
+}
+
++ (Class)stepViewControllerClass {
+    return [ORKReviewStepViewController class];
+}
+
+- (nonnull instancetype)initWithCoder:(NSCoder *)aDecoder {
+    self = [super initWithCoder:aDecoder];
+    if (self) {
+        ORK_DECODE_OBJ_CLASS(aDecoder, steps, NSArray);
+        ORK_DECODE_OBJ(aDecoder, resultSource);
+    }
+    return self;
+}
+
+- (void)encodeWithCoder:(NSCoder *)aCoder {
+    [super encodeWithCoder:aCoder];
+    ORK_ENCODE_OBJ(aCoder, steps);
+    ORK_ENCODE_OBJ(aCoder, resultSource);
```

Comment:
@umerkhan-apple Just want to make sure that we're on the same page here: `isStandalone` is a computed property, so it's basically just a shorthand for `_steps != nil`. So do we need an extra encoding/decoding step in this instance? If so, how should we assign any value to this read-only property? Finally, we won't be able to use the `ORK_ENCODE` / `ORK_DECODE` makros since there is no associated ivar.